integration-tests/virtual-threads/vertx-event-bus-virtual-threads/src/test/java/io/quarkus/virtual/vertx/NoPinningVerify.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.virtual.vertx;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * An integration test reading the output of the unit test to verify that no tests where pinning the carrier thread.
```

Comment:
Is this a common pattern we use? Reading the XML files generated by surefire?

Reply:
Unfortunately at the moment, yes...
I attempted multiple times to detect it but it's not very reliable. 

My next idea is to use byte buddy to manipulate JDK code, but it won't work in native.

Reply:
Ok, I see.

Reply:
I always thought you were a SAX lover :-D